test/helpers.h

```diff
@@ -119,4 +119,15 @@ class PreprocessorHelper
 #define DINIT_NOEXCEPT
 #endif
 
+// Default construct object to avoid bug in clang
+// error: default member initializer for 'y' needed within definition of enclosing class 'X' outside of member functions
+struct make_default_obj
```

Comment:
hmm spontanously it feels strange we need a `make_default_obj`. I thought the syntax `const Fred& fred={}` was normal.

Reply:
It works around an bug in the compiler (see https://stackoverflow.com/questions/53408962) - something I already did with my initial workaround but @pfultz2 compiler is somehow acting up.

Reply:
> I thought the syntax const Fred& fred={} was normal.

Yes that is correct but there is a bug in clang which fails to compile. Even `Fred = Fred{}` fails to compile in this context. This is the only way I found to get it to work.

Reply:
> something I already did with my initial workaround but @pfultz2 compiler is somehow acting up.

The compiler is clang 13 on ubuntu 20.04. I dont really see how adding `noexcept` would fix this issue.

Reply:
I am using Clang 10 to 18 from `apt.llvm.org` on ubuntu 20.04.

I just tested your changes locally and it also works around the issue. Please also remove the `DINIT_NOEXCEPT` stuff (which is no longer necessary) and then we can merge this.